src/app/icd/client/DefaultICDClientStorage.h

```diff
@@ -52,6 +52,17 @@ class DefaultICDClientStorage : public ICDClientStorage
 
     ICDClientInfoIterator * IterateICDClientInfo() override;
 
+    /**
+     * When decrypting check-in messages, the system needs to iterate through all keys
```

Comment:
Thank you, this comment finally makes the API make sense.

Note that this also makes it clearer that this storage class is not appropriate for settings where fabric indices are not stable (Darwin, Google's integrations as I understand them, pretty much every non-toy client I am aware of).  That would be a good restriction to document at the top of this file in the general "what is this class for?" documentation.

Reply:
done, thanks